### PR TITLE
Scanner.ts

### PR DESCRIPTION
**Pull request by lsaether**
depends on #51

### CODE REVIEW

File: src/scanner.ts [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+  getLeftBlock(latest) {
+    const leftBlock = latest - this.config.scanSpread;
+    return leftBlock < 0 ? 0 : leftBlock;
+  }
+
+  getRightTimestamp(leftTimestamp, latestTimestamp) {
+    return 2 * latestTimestamp - leftTimestamp;
+  }
+
+  // @param request {Object} of form {address: 0xAF...34, uintArgs: uint[12]}
```
Comment (Kuzirashi) [inline src/scanner.ts]: Unnecessary comment.

File: src/scanner.ts [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+    //   await this.requestFactory.stopWatch(this.requestWatcher);
+    //   this.log.info('Watching STOPPED');
+    // }
+
+    // Mark that we've stopped.
+    this.running = false;
+    this.config.logger.info('Scanner STOPPED');
+  }
+
+  // isValidBlock(block) {
```
Comment (Kuzirashi) [inline src/scanner.ts]: Unnecessary comment.

File: src/scanner.ts [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+    this.running = true;
+    this.config.logger.info('Scanner STARTED');
+  }
+
+  async stop() {
+    // Clear scanning intervasls.
+    clearInterval(this.cacheScanner);
+    clearInterval(this.chainScanner);
+
+    // if (this.requestWatcher) {
```
Comment (Kuzirashi) [inline src/scanner.ts]: Unnecessary comment.

File: src/scanner.ts (hunk 1 above)
Comment (lsaether) [reply src/scanner.ts]: :100:

File: src/scanner.ts (hunk 2 above)
Comment (lsaether) [reply src/scanner.ts]: :100:

File: src/scanner.ts (hunk 3 above)
Comment (lsaether) [reply src/scanner.ts]: :100: * 3

**Pull request closed without merging**